jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -1106,7 +1107,32 @@ static Chunk next(Chunk chunk)
             if (chunk == null)
                 return null;
             if (Content.Chunk.isFailure(chunk))
-                return chunk.isLast() ? chunk : null;
+            {
+                // handle transient failure
+                if (!chunk.isLast())
+                    return null;
+
+                // handle persistent failure with lazy copy to avoid returning the same exception.
+                return new Empty()
+                {
+                    private transient Throwable failure;
```

Comment:
Not `transient`.

I'd make it `final`, as the only thing to do with a failure chunk is to call `getFailure()`, so lazyness is not necessary.

I would just simplify this by calling `Chunk.from(copyOf(chunk.getFailure()))`.

Reply:
This is a static method, so we can't have a boolean here.   

Besides, the first chunk has already been returned and we are being asked for the next chunk here.  So the "boolean" is always `true`.  I.e. we have already returned the failure once and any use of the next chunk is the second usage (`true` already).

Any state we remember has to be in the Chunk that we return.   If we are passed in a non-transient failure Chunk, then we return a Chunk that IF AND ONLY IF it is used, will create another failure exception.

Reply:
Ok re lazyness, but remove `transient` from the field modifiers.
`transient` is for Java serialization, and we don't care here.